### PR TITLE
Edit urls

### PR DESCRIPTION
**Pull request by tejaswinichile**
Issue #246

### CODE REVIEW

**Comment by breezyboa** [comment]
@tejaswinichile _a some tests are failing. can you take care of it

**Comment by breezyboa** [comment]
@tejaswinichile _a can you please resolve conflicts

**Comment by tejaswinichile** [comment]
@sudeeptarlekar  _a

File: app/javascript/Admin/Admin/Request/Url.elm [hunk 1]
```diff
@@ -26,3 +26,25 @@ createUrl =
                 GQLBuilder.request createUrlInput createUrlMutation
             )
         )
+
+
+updateUrl : Reader ( NodeEnv, UrlData ) (Task GQLClient.Error UrlData)
+updateUrl =
+    Reader.Reader
+        (\( nodeEnv, urlData ) ->
+            (GQLClient.sendMutation (graphqlUrl nodeEnv) <|
```
Comment (sudeeptarlekar) [inline app/javascript/Admin/Admin/Request/Url.elm]: will this work on staging?

File: app/javascript/Admin/Admin/Request/Url.elm [hunk 2]
```diff
@@ -26,3 +26,25 @@ createUrl =
...
+                GQLBuilder.request urlData updateUrlMutation
+            )
+        )
+
+
+requestUrlById : UrlId -> Reader ( NodeEnv, ApiKey ) (Task GQLClient.Error UrlData)
+requestUrlById urlId =
+    Reader.Reader
+        (\( env, apiKey ) ->
+            GQLClient.sendQuery (graphqlUrl env) <|
```
Comment (sudeeptarlekar) [inline app/javascript/Admin/Admin/Request/Url.elm]: remove API key from params if you are not using it

File: app/javascript/Admin/Route.elm [hunk 3]
```diff
@@ -39,6 +40,7 @@ routeMatcher =
         , Url.map UrlCreate (s "urls" </> s "new")
         , Url.map CategoryCreate (s "categories" </> s "new")
         , Url.map ArticleEdit (s "articles" </> string)
+        , Url.map UrlEdit (s "admin" </> s "urls" </> string </> s "edit")
```
Comment (sudeeptarlekar) [inline app/javascript/Admin/Route.elm]: Please don't use admin namespace

File: app/javascript/Admin/Admin/Request/Url.elm (hunk 1 above)
Comment (tejaswinichile) [reply app/javascript/Admin/Admin/Request/Url.elm]: Right, I need to use `customSendMutation` there. 👍 I'll make that change.

File: app/graphql/mutations/url_mutations.rb [hunk 4]
```diff
@@ -34,7 +34,7 @@ class Mutations::UrlMutations
       input_field :url, !types.String
     end
     input_field :id, !types.String
-    input_field :url, !UrlInputObjectType
+    input_field :url, !types.String
```
Comment (sudeeptarlekar) [inline app/graphql/mutations/url_mutations.rb]: if we are not using `UrlInput` then pls remove it...

**Pull request merged**